daisy_workflows/image_build/windows_for_containers/container_image_install.ps1

```diff
@@ -47,61 +47,163 @@ function Get-MetadataValue {
+  # Windows Server 2019 does not support the nanoserver container image, only
+  # Servercore:
+  # https://blogs.technet.microsoft.com/virtualization/2018/11/13/windows-server-2019-now-available/.
+  if ($windows_version -eq '2019') {
+    return $false
+  }
+  return $true
+}
 
-    Write-Host 'Restarting computer to finish install'
-    Restart-Computer -Force
+function Get-ServerCoreImageName {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+
+  $repo = Get-ContainerRepo $windows_version
+  if ($windows_version -eq '1709' -or $windows_version -eq '1803') {
+    $image = 'windowsservercore'
+  } else {
+    $image = 'servercore'
   }
-  else {
-    # For some reason the docker service may not be started automatically on the
-    # first reboot, although it seems to work fine on subsequent reboots. The
-    # docker service must be running or else the vEthernet interface may not be
-    # present.
-    Restart-Service docker
-
-    Write-Host 'Setting host vEthernet MTU to 1460'
-    Get-NetAdapter | Where-Object {$_.Name -like 'vEthernet*'} | ForEach-Object {
-      & netsh interface ipv4 set subinterface $_.InterfaceIndex mtu=1460 store=persistent
-    }
+  $label = Get-ContainerVersionLabel $windows_version
+  return "${repo}/${image}:${label}"
+}
 
-    # As most if not all Windows containers are based on one of these images
-    # we pull it here so that running a container using this image is quick.
-    Write-Host 'Pulling latest Windows containers'
-    & docker pull "microsoft/windowsservercore:${windows_version}"
-    if (!$?) {
-      throw "Error running 'docker pull microsoft/windowsservercore:${windows_version}'"
-    }
-    & docker pull "microsoft/nanoserver:${windows_version}"
-    if (!$?) {
-      throw "Error running 'docker pull microsoft/nanoserver:${windows_version}'"
-    }
+function Get-NanoserverImageName {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+  $repo = Get-ContainerRepo $windows_version
+  $image = 'nanoserver'
+  $label = Get-ContainerVersionLabel $windows_version
+  return "${repo}/${image}:${label}"
+}
+
+function Get-BaseContainerImageNames {
+  param (
+    [parameter(Mandatory=$true)]
+      [string]$windows_version
+  )
+  $images = @(Get-ServerCoreImageName $windows_version)
+  if (Supports-NanoserverContainerImage $windows_version) {
+    $images += (Get-NanoserverImageName $windows_version)
+  }
+  return $images
+}
+
+function Run-FirstBootSteps {
+  Write-Host 'Installing NuGet module'
+  Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force
+
+  Write-Host 'Installing DockerMsftProvider module'
+  Install-Module -Name DockerMsftProvider -Repository PSGallery -Force
+
+  $docker_version = "18.09"
+  Write-Host "Installing Docker EE ${docker_version}"
+  Install-Package -Name docker -ProviderName DockerMsftProvider -Force -RequiredVersion ${docker_version}
+
+  Write-Host 'Enabling IPv6'
```

Comment:
I do not believe we need this anymore as the base images don't disable ipv6

Reply:
Ok, added a commit that removes this.